client/solve.go

```diff
@@ -178,6 +179,11 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
 		opt.FrontendAttrs[k] = v
 	}
 
+	// if OCILayout content.Store options are defined, make them available to the builder
+	if len(opt.OCILayouts) > 0 {
+		s.Allow(sessioncontent.NewAttachable(opt.OCILayouts))
```

Comment:
This needs to be connected with the store map used for local cache. Othewise you are registering the same service twice.

Reply:
Ah does a.Allow() only have one service of each type?